Package dependencies ensure the correct library is linked at run-time.
Remove overly restictive version check.

diff -Naur --exclude=debian perl-5.10.0.orig/ext/DB_File/version.c perl-5.10.0/ext/DB_File/version.c
--- perl-5.10.0.orig/ext/DB_File/version.c	2007-11-18 02:26:25.000000000 +1100
+++ perl-5.10.0/ext/DB_File/version.c	2007-11-26 07:13:26.000000000 +1100
@@ -48,12 +48,14 @@
 
     (void)db_version(&Major, &Minor, &Patch) ;
 
+#ifndef DEBIAN
     /* Check that the versions of db.h and libdb.a are the same */
     if (Major != DB_VERSION_MAJOR || Minor != DB_VERSION_MINOR 
 		|| Patch != DB_VERSION_PATCH)
 	croak("\nDB_File needs compatible versions of libdb & db.h\n\tyou have db.h version %d.%d.%d and libdb version %d.%d.%d\n",  
 		DB_VERSION_MAJOR, DB_VERSION_MINOR, DB_VERSION_PATCH, 
 		Major, Minor, Patch) ;
+#endif /* DEBIAN */
     
     /* check that libdb is recent enough  -- we need 2.3.4 or greater */
     if (Major == 2 && (Minor < 3 || (Minor ==  3 && Patch < 4)))
